pkg/operator/resources/batchapi/errors.go

```diff
@@ -0,0 +1,118 @@
+	ErrFailedToEnqueueMessages    = "batchapi.failed_to_enqueue_messages"
+	ErrMessageExceedsMaxSize      = "batchapi.message_exceeds_max_size"
+	ErrConflictingFields          = "batchapi.conflicting_fields"
+	ErrBatchItemSizeExceedsLimit  = "batchapi.item_size_exceeds_limit"
+	ErrSpecifyExactlyOneKey       = "batchapi.specify_exactly_one_key"
+)
+
+func ErrorBatchAPINotDeployed(apiName string) error {
+	return errors.WithStack(&errors.Error{
+		Kind:    ErrBatchAPINotDeployed,
+		Message: fmt.Sprintf("BatchAPI api named '%s' is not deployed", apiName),
+	})
+}
+
+func ErrorJobNotFound(jobKey spec.JobKey) error {
+	return errors.WithStack(&errors.Error{
+		Kind:    ErrJobNotFound,
+		Message: fmt.Sprintf("unable to find job %s", jobKey.UserString()),
+	})
+}
+
+func ErrorJobIsNotInProgress() error {
+	return errors.WithStack(&errors.Error{
+		Kind:    ErrJobIsNotInProgress,
+		Message: "cannot stop job because it is not in progress",
+	})
+}
+
+func ErrorJobHasAlreadyBeenStopped() error {
+	return errors.WithStack(&errors.Error{
+		Kind:    ErrJobHasAlreadyBeenStopped,
+		Message: "job has already been stopped",
+	})
+}
+
+func ErrorNoS3FilesFound() error {
+	return errors.WithStack(&errors.Error{
+		Kind:    ErrNoS3FilesFound,
+		Message: "no s3 files found based on search criteria",
+	})
+}
+
+func ErrorNoDataFoundInJobSubmission() error {
+	return errors.WithStack(&errors.Error{
+		Kind:    ErrNoDataFoundInJobSubmission,
+		Message: "unable to enqueue batches because no data was found",
+	})
+}
+
+func ErrorFailedToEnqueueMessages(message string) error {
+	return errors.WithStack(&errors.Error{
+		Kind:    ErrFailedToEnqueueMessages,
+		Message: fmt.Sprintf(message),
+	})
+}
+
+func ErrorMessageExceedsMaxSize(messageSize int, messageLimit int) error {
+	return errors.WithStack(&errors.Error{
+		Kind:    ErrMessageExceedsMaxSize,
+		Message: fmt.Sprintf("cannot enqueue message because its size of %d bytes exceeds the %d bytes limit; use a smaller batch size or reduce the size of each of item in the batch", messageSize, messageLimit),
+	})
+}
+
+func ErrorConflictingFields(key string, keys ...string) error {
+	allKeys := append(keys, key)
+	return errors.WithStack(&errors.Error{
+		Kind:    ErrConflictingFields,
+		Message: fmt.Sprintf("please specify either the %s field (both not more than one at the same time)", s.StrsOr(allKeys)),
+	})
+}
+
+func ErrorItemSizeExceedsLimit(index int, size int, limit int) error {
+	return errors.WithStack(&errors.Error{
+		Kind:    ErrBatchItemSizeExceedsLimit,
+		Message: fmt.Sprintf("item %d has size %d bytes which exceeds the limit %d", index, size, limit),
+	})
+}
+
+func ErrorSpecifyExactlyOneKey(key string, keys ...string) error {
+	allKeys := append(keys, key)
+	return errors.WithStack(&errors.Error{
+		Kind:    ErrSpecifyExactlyOneKey,
+		Message: fmt.Sprintf("specify exactly one of the following keys %s", s.StrsOr(allKeys)), // TODO add job specification documentation
```

Comment:
nit: either "specify exactly one of the following keys: %s" or "specify exactly one of %s"